### PR TITLE
fix(rest): missing user-agent separator

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11473)

### CODE REVIEW

File: google/cloud/internal/curl_impl.cc [hunk 1]
```diff
@@ -212,8 +212,9 @@ CurlImpl::CurlImpl(CurlHandle handle,
   socket_options_.send_buffer_size_ =
       options.get<MaximumCurlSocketSendSizeOption>();
 
-  auto const& agents = options.get<UserAgentProductsOption>();
-  user_agent_ = absl::StrCat(absl::StrJoin(agents, " "), UserAgentSuffix());
+  auto agents = options.get<UserAgentProductsOption>();
+  agents.push_back(UserAgentSuffix());
+  user_agent_ = absl::StrCat(absl::StrJoin(agents, " "));
```
Comment (dbolduc) [inline google/cloud/internal/curl_impl.cc:217]: Drop the `absl::StrCat` with one argument?

File: google/cloud/storage/tests/curl_request_integration_test.cc [hunk 2]
```diff
@@ -357,11 +357,13 @@ TEST(CurlRequestTest, UserAgent) {
...
-  ASSERT_EQ(1U, payload.count("headers"));
-  auto headers = payload["headers"];
-  EXPECT_THAT(headers.value("User-Agent", ""),
-              HasSubstr("test-user-agent-prefix"));
-  EXPECT_THAT(headers.value("User-Agent", ""), HasSubstr("gcloud-cpp/"));
+  ASSERT_TRUE(payload.is_object()) << "payload=" << response->payload;
+  ASSERT_TRUE(payload.contains("headers")) << "payload=" << response->payload;
+  auto const products = std::vector<std::string>(
+      absl::StrSplit(payload["headers"].value("User-Agent", ""), ' '));
+  EXPECT_THAT(products, AllOf(Contains("test-user-agent-prefix-1"),
```
Comment (dbolduc) [inline google/cloud/storage/tests/curl_request_integration_test.cc:364]: comment: I don't know what gives nicer errors between `AllOf(Contains(m1), Contains(m2), Contains(m3))` and `IsSupersetOf({m1, m2, m3})`

**Review by dbolduc — APPROVED** [review]

File: google/cloud/internal/curl_impl.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/internal/curl_impl.cc]: Thanks, fixed.

File: google/cloud/storage/tests/curl_request_integration_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/tests/curl_request_integration_test.cc:364]: I did not want to fiddle with the type of `{"foo", "bar", StartsWith("baz")}`, or `{Eq("foo"), Eq("bar"), StartsWith("baz")}`.  I think both are ambiguous (they need to be `initializer_list<T>` and `T` can be too many things.

**Pull request merged**